ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/builders/ExtensionsErrorReporter.java

```diff
@@ -597,6 +600,37 @@ private boolean resourceExists(String location) {
 			}
 		}
 
+		return isIgnoredResourceUri(location);
+	}
+
+	private boolean isIgnoredResourceUri(String location) {
+		final var ignoredProtocols = CompilerFlags.getString(fProject, CompilerFlags.P_IGNORED_RESOURCE_PROTOCOLS);
+		final var st = new StringTokenizer(ignoredProtocols, ","); //$NON-NLS-1$
+		if (!st.hasMoreTokens()) {
+			return false;
+		}
+		try {
+			final var uri = new URI(location);
+			if (uri.isAbsolute()) {
+				// Exclude syntactically valid file paths that are also
+				// syntactically valid URIs, such as "C:/somePath" where
+				// "C" would be interpreted as URI protocol:
+				try {
+					java.nio.file.Path.of(location);
+				} catch (java.nio.file.InvalidPathException e) {
+					// OK, location does not match a file path
+					final var scheme = uri.getScheme();
+					while (st.hasMoreElements()) {
+						String protocol = st.nextToken().trim();
+						if (scheme.equalsIgnoreCase(protocol)) {
+							return true;
+						}
+					}
+				}
+			}
+		} catch (URISyntaxException e) {
+			// location is not a valid URI
+		}
 		return false;
```

Comment:
As [RFC 2396](https://www.rfc-editor.org/rfc/rfc2396#section-3) specifies that the scheme must be followed by a colon (for absolute URIs) I think this entire method can be simplified to just
```suggestion
		String ignoredProtocols = CompilerFlags.getString(fProject, CompilerFlags.P_IGNORED_RESOURCE_PROTOCOLS);
		if (ignoredProtocols.isEmpty()) {
			return false;
		}
		return Arrays.stream(ignoredProtocols.split(",")).map(p -> p + ":").anyMatch(location::startsWith); //$NON-NLS-1$ //$NON-NLS-2$
```

I.e. it just checks if the location starts with any protocol plus a colon.

Reply:
There are two issues with this alternative solution: (1) It doesn't trim any leading or trailing spaces (and this seems to be done everywhere else) and (2) it doesn't do a case-insensitive match (protocol comparison is defined to be case-insensitive). So I'm working on a modified approach now.

Reply:
That's both correct.
But your modifications to refine this approach look good. :)